provision_certificate/provision_certificate.py

```diff
@@ -0,0 +1,75 @@
+import requests
+import os
+import sys
+import json
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+
+STRIKETRACKER_URL = os.environ['STRIKETRACKER_URL'] if 'STRIKETRACKER_URL' in os.environ else 'https://striketracker.highwinds.com'
+if len(sys.argv) != 6:
+    print "Usage: python provision_certificate.py [account_hash] [host name] [certificate] [key] [caBundle]"
+    sys.exit()
+PARENT_ACCOUNT = sys.argv[1]
+HOST_NAME = sys.argv[2] # Friendly name for host in StrikeTracker
+CERTIFICATE = sys.argv[3]  # SSL certificate file
+KEY = sys.argv[4]  # SSL certificate key file
+CA_BUNDLE = sys.argv[5]  # SSL certificate bundle file
+OAUTH_TOKEN = os.environ['STRIKETRACKER_TOKEN']
+
+
+# List certificates
+def list_certificates():
+    certificates_response = requests.get(
+        STRIKETRACKER_URL + "/api/accounts/{accountHash}/certificates".format(accountHash=PARENT_ACCOUNT),
+        headers={"Authorization": "Bearer %s" % OAUTH_TOKEN, "Content-Type": "application/json"})
+    return certificates_response.json()
+
+
+# Get certificate by hostname
+def get_certificate(hostname):
```

Comment:
The idea was the script tries to find a certificate by it's commonName and if not found create it.
In case of certificate id it's not clear how to behave in such a situation. Split the script into parts?
Also i don't have any clue how HWCDN selects certificate if there are several with the same commonName.